src/components/CriteriaProvider/CriteriaProvider.tsx

```diff
@@ -154,6 +174,33 @@ export const CriteriaProvider: FC<CriteriaProviderProps> = memo(({ children }) =
             if (newCriteria) setCriteria(currentCriteria => [...currentCriteria, ...newCriteria]);
           } else if (file.name.endsWith('.cql')) {
             addCqlCriteria(rawContent);
+          } else if (file.name.endsWith('.zip')) {
+            const cqlObj: CqlObject = { main: '', libraries: {} };
+            const zip = await JSZip.loadAsync(file);
+            const zipFiles = Object.values(zip.files);
+            // Check for criteria in each of the cql files, add cql to libraries if no criteria
+            for (let i = 0; i < zipFiles.length; i++) {
+              const zipFile = zipFiles[i];
+
+              // Skip files that do not end with .cql
+              if (!zipFile.name.endsWith('.cql')) continue;
+
+              const fileContents = await zipFile.async('string');
+              if (CQL_HELPER_LIBRARIES.includes(zipFile.name)) {
+                cqlObj.libraries[zipFile.name] = { cql: fileContents };
```

Comment:
I guess I'm a little confused - does the AT always produce only one cql? If someone uploads their own cql zip library this would fail. I guess I presumed the cql for the helper libraries would be known by the builder and then it would import the cql for any other files which are not in the list of helper files. Perhaps that is out of scope though

Reply:
yea previous approach where I was separating out a "main" library and helper libraries was making a lot of assumptions. I updated the code to implement the approach Dylan mentioned in https://github.com/mcode/pathway-builder/pull/62#discussion_r518284866 so it should work even if we don't know the name of the helper libraries.